src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -423,9 +426,91 @@ export class StudyViewPageStore {
         this._comparisonGroupsChangeCount += 1;
     }
 
-    private createPieChartComparisonSession(
+    private async createNumberAttributeComparisonSession(
         clinicalAttribute:ClinicalAttribute,
-        clinicalAttributeValues:{ value:string, color:string}[],
+        bins:DataBin[],
+        statusCallback:(phase:LoadingPhase)=>void
+    ) {
+        statusCallback(LoadingPhase.DOWNLOADING_GROUPS);
+        return new Promise<string>((resolve)=>{
+            onMobxPromise(this.selectedSamples,
+                async (selectedSamples)=>{
+                    // get clinical data for the given attribute
+                    const entityIdKey = (clinicalAttribute.patientAttribute ? "patientId" : "sampleId")
+                    const data = await client.fetchClinicalDataUsingPOST({
+                        clinicalDataType: clinicalAttribute.patientAttribute ? "PATIENT" : "SAMPLE",
+                        clinicalDataMultiStudyFilter: {
+                            attributeIds: [clinicalAttribute.clinicalAttributeId],
+                            identifiers: selectedSamples.map(s=>({ studyId: s.studyId, entityId: s[entityIdKey] }))
+                        }
+                    });
+
+                    // copy bins, take out NA bin if there is one
+                    let binsWithData = _.cloneDeep(bins)
+                        .map(bin=>{
+                            (bin as any).data = [];
+                            return bin as DataBinWithData;
+                        });
+
+                    const naBinIndex = binsWithData.findIndex(bin=>bin.specialValue === "NA");
```

Comment:
You need to find all nonnumerical values. The bar chart supports string.
For instance, Tumor Purity in https://cbioportal.mskcc.org/study?id=mskimpact
Also some special ranges, for instance '<18'  in genie `Age at Which Sequencing was Reported`. http://genie.cbioportal.org/study?id=genie_public
@jjgao @adamabeshouse this is basically redoing all the logics at the server side. @onursumer can we make a API call for this? Give a clinical attribute id, study view filter, and return the sample ids in every data bin.

Reply:
that would be much much much better. sample ids as well as the name, so frontend doesnt have to combine "specialValue" and endpoint etc.